flowcraft/generator/engine.py

```diff
@@ -1524,6 +1525,46 @@ def export_directives(self):
         # Flush params json to stdout
         sys.stdout.write(json.dumps(directives_json))
 
+    def fetch_docker_tags(self):
+        """
+        Export all dockerhub tags associated with each component given by
+        the -t flag.
+        """
+
+        # starts an array with the headers
+        tags_list = [
+            ["component", "container", "tags"],
+        ]
+
+        # Skip first init process and iterate through the others
+        for p in self.processes[1:]:
+            template = p.template
+            # fetch repo name from directives of the template
+            repo = p.directives[template]["container"]
+            # make the request to docker hub
+            r = requests.get(
+                "https://hub.docker.com/v2/repositories/{}/tags/".format(
+                    repo
+                ))
+            # checks the status code of the request, if it is 200 then parses
+            # docker hub entry, otherwise retrieve no tags but alerts the user
+            if r.status_code != 404:
+                # parse response content to dict and fetch results key
+                r_content = json.loads(r.content)["results"]
+                for version in r_content:
+                    tags_list.append([template, repo, version["name"]])
+            else:
+                tags_list.append([template, repo, "No DockerHub tags"])
+
+        # iterate through each entry in tags_list and print the list of tags
+        # for each component
+        for x, entry in enumerate(tags_list):
+            # adds different color to the header in the first list
+            color = "white" if x > 0 else "purple_bold"
+            sys.stdout.write(
+                colored_print("{}\n".format("\t".join(entry)), color)
```

Comment:
I would suggest something better to represent text in a tabular format using string formatting:

```
sys.stdout.write(
    colored_print("{0:15.15} {1: ^20} {2: ^20}".format(*entry), color)
)
```

This will put the first element in the first 15 characters, left aligned, and the other two elements with 20 characters, right aligned. You can tweak the size of each field according to your tests to see which will have the best presentation.

Reply:
Wow that is neat. I will make a couple of tests with it.